### ISSUE TITLE
Series 4 Wiki Strings Initial Housekeeping

### ISSUE THREAD

**Issue by mattodd** [OpenSourceMalaria/OSM_To_Do_List#290]
Hi Kevin ( @kzha9011 ) . Thanks very much for agreeing to help the OSM community with some wiki maintenance. 

OK, so the first thing we need to do is to make sure that the data listed in the Strings section at the bottom of the wiki accurately reflects what’s in the wiki page.

http://openwetware.org/wiki/OpenSourceMalaria:Triazolopyrazine_%28TP%29_Series

This is a fair bit of work, so let’s break it up a little into chunks.

There are lots of compounds on the page. You’ll need structures for those. Many of them can be found in Chemdraw files attached to this page

http://malaria.ourexperiment.org/osdd_malaria_shared/8585/Files_Associated_with_Wiki_Page_on_Series_4.html

or the Github area for the first Series 4 paper

https://github.com/OpenSourceMalaria/OSMSeries4Paper1

i.e. you should be able to just download those Chemdraw files and open them up in Chemsketch, is that right?

Assuming you can (let me know if not, in which case we’ll need a workaround) then that will save you some time vs drawing them all out again! 

But what’s needed is this:

For each compound listed in the Strings section we need data using this pattern:

MMV code [space] Smiles [space] InChI [space] InChiKey [space] potency in micro molar

The entries for the first two compounds (MMV639565 and MMV639725) are already done. We need this replicated for all the other compounds. Some compounds have some data there already. Some compounds have none. None of the other compounds have potencies, I don’t think (the potencies are all listed in the wiki page somewhere). There are a few compounds down the bottom lacking MMV numbers - don’t worry about that, we still need the strings. It’s important to replicate the two completed entries exactly, so that we can easily have software come along and extract information from this section.

How about making a start, doing 2-3 entries and then letting me see how you’re getting on?

For adding text to the wiki, copying and pasting directly into the wiki source page ought to work fine, since the text comes over without formatting, but be sure to save your page regularly in case of trouble. We can always revert to an earlier version of the page is something goes wrong.

Try to use this platform for queries, rather than email, but remembering that anything written here is public. The same applies if you reply to emails from the Github platform - they go public!

**Comment by miike** [comment]
Quick sidenote: Instead of a space between fields we should aim to use a tab (as long as the wiki preserves this). If we use a space there's a chance we won't be able to split a row into the correct number of fields if one value contains a space, tabs will eliminate this problem.

**Comment by drc007** [comment]
Please use tabs not spaces. I suspect all cheminformatics tools will expect tab as the delimiter. Also can the page be saved as plain text, i.e. no formatting or styling.

**Comment by mattodd** [comment]
Doesn't look like it accepts tabs. When I press the tab key the cursor just moves to the next available window, rather than moves forward. Anyone know if I ought to be inserting a character that acts as a proxy for Tab? What's used in wikipedia or other wikis?

**Comment by drc007** [comment]
It might be better to create a plain text file locally, and then simply upload the file?

**Comment by mattodd** [comment]
Markdown file hosted on Github would do the trick? I may have the technical
capacity for that now, if so.

On 1 April 2015 at 21:36, Chris Swain <notifications@github.com> wrote:

>  It might be better to create a plain text file locally, and then simply
> upload the file?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-88434797>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
Or you could use a Google docs spreadsheet which would allow collaborative editing. You could then export as a plain text tab delimited file.

**Comment by miike** [comment]
Text file in the repo would be ideal and that could easily be updated by exporting a Google doc as @drc007 mentioned.

**Comment by mattodd** [comment]
You mean like this?

https://github.com/OpenSourceMalaria/OSM_compounds/blob/master/Series4-SDF-lite

I used "tab" between the various components. Working?

Lay it out for me - how does the communication work between GH and Google
Docs here? I could export this from GH to some generic file type that is
readable by GDocs (as a sheet?) and I can export back from GDocs to
markdown (and therefore maintain a master file on my laptop?)

On 1 April 2015 at 21:53, miike <notifications@github.com> wrote:

>  Text file in the repo would be ideal and that could easily be updated by
> exporting a Google doc as @drc007 <https://github.com/drc007> mentioned.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-88438539>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
I guess it depends on how comfortable people will be to edit this file directly? If that is the plan?

https://github.com/OpenSourceMalaria/OSM_compounds/blob/master/Series4-SDF-lite

**Comment by mattodd** [comment]
Well, if there are import/export options available then people could edit
the Google Doc and that could be transferred to GH. But editing the GH file
directly would be best. It's pretty easy if it's just text. A full SDF that
way would be tough, but this is just a summary of structures and potencies.
Doable?

On 1 April 2015 at 22:40, Chris Swain <notifications@github.com> wrote:

>  I guess it depends on how comfortable people will be to edit this file
> directly? If that is the plan?
>
>
> https://github.com/OpenSourceMalaria/OSM_compounds/blob/master/Series4-SDF-lite
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-88447807>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
It is workable but you may need to have some training for editors.
A couple of points to note, it would be helpful to have a header row labelling each column, will you be adding many more fields? As you add more fields it will get increasingly difficult to ensure the correct data goes into the correct place, which is why a spreadsheet is useful.

**Comment by mattodd** [comment]
I wasn't planning on more fields, but that depends what people might want
to do. Keeping it simple as possible (i.e. not an SDF) I was thinking the
three strings, the compound ID, any synonyms and the potency. This would
allow simple SAR-type analyses. Ought we to include something else that is
seen to be crucial in the short term, do you think?

On 1 April 2015 at 22:49, Chris Swain <notifications@github.com> wrote:

>  It is workable but you may need to have some training for editors.
> A couple of points to note, it would be helpful to have a header row
> labelling each column, will you be adding many more fields? As you add more
> fields it will get increasingly difficult to ensure the correct data goes
> into the correct place, which is why a spreadsheet is useful.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-88448860>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
Let's go with this to get all the data together. If you want to add more fields we can simply import this into whatever format/application you choose.

**Comment by mattodd** [comment]
OK, @kzha9011 I recommend we keep going with the wiki as per my first post in this thread, and then we port over to a file on Github a little later.

**Comment by kzha9011** [comment]
Only just got home but I'll get down to work asap. I'm a bit confused. I'm guessing that I should use tabs and that I do the work directly on the wiki and not on google docs

**Comment by drc007** [comment]
How will you port to GH file? You don't want to have to copy and paste each item of data.

**Comment by mattodd** [comment]
Yes - on the wiki, just use spaces for now (tabs don't work) and just input the stuff I just mentioned to Chris. *Unless* you feel comfortable directly editing the text file on Github that I linked to earlier, i.e. 
https://github.com/OpenSourceMalaria/OSM_compounds/blob/master/Series4-SDF-lite

**Comment by miike** [comment]
I'd recommend leaving the wiki for the moment (as it's not great for text editing) and instead edit either directly on Github or in Google Docs. It's pretty easy to export in Google docs to a tab delimited format (File->Download as->Tab delimited values) and then copy this directly into the "SDF-lite" file via [direct edit](https://github.com/OpenSourceMalaria/OSM_compounds/edit/master/Series4-SDF-lite)

**Comment by kzha9011** [comment]
Hi guys
Run into a bump. Can't access the chemdraw pics on chemsketch. Not sure what I can do. I don't think drawing them is a possibility. I have no idea how to use the software

**Comment by mattodd** [comment]
OK, so what's needed is free software that is not Chemdraw or Chemsketch, which can be used to read .cdx files and be used to generate SMILES, InChI and InChiKey strings. Any ideas from anyone?

**Comment by cdsouthan** [comment]
Try  http://openbabel.org/wiki/Main_Page  its a useful all purpose converter anyway

**Comment by drc007** [comment]
Have a look at MarvinSketch https://www.chemaxon.com/products/marvin/, I think the standalone is free.

**Comment by mattodd** [comment]
Thanks guys. @kzha9011 want to take a look? Sounds like these would do the trick. If not, let us know what the barrier is. If you can get the strings for these compounds, do you want to try the wiki or go full-on for the Github file (which would probably be more immediately useful).

**Comment by drc007** [comment]
Have a look at https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/286

**Comment by drc007** [comment]
Have a look at https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/285

**Comment by mattodd** [comment]
Hi @jlee8294, thanks for agreeing to try this out! It's important that we have a good workflow for people to add new OSM molecules to a central database of structures.

So. Here are the steps we need you to follow.

1. Go to the wiki: http://openwetware.org/wiki/OpenSourceMalaria:Triazolopyrazine_%28TP%29_Series and pick a molecule in the "strings" section that has no strings listed.
2. Check the new spreadsheet @drc007 has been working on (http://www.cheminfo.org/Chemistry/Parsing%20data/Tab_delimited_Parallel_Coordinates.html?tsvURL=http%3A%2F%2Fgoogledocs.cheminfo.org%2Fspreadsheets%2Fd%2F1Rvy6OiM291d1GN_cyT6eSw_C3lSuJ1jaR7AJa8hgGsc%2Fexport%3Fformat%3Dtsv) to see if the strings have already been calculated. If they have (e.g. for MMV669001), paste them from the Google Sheet into the wiki (you'll need to get an OpenWetWare account and click "edit" at the top of the Strings section and add the data into this section in the way that is shown for the molecules already there.) It'd be good to have this section complete.
3. If the molecule's in the wiki strings section but not in the Google sheet, you'll need to find a file where the molecule appears to that you can generate the strings. These will be Chemdraw files, with the .cdx file extension. Many of these files are on the lab notebook page containing the wiki picture files here: http://malaria.ourexperiment.org/osdd_malaria_shared/8585. But if the molecule you're interested in has an OSM number that has been converted on the wiki into a hyperlink (e.g. OSM-S-201) then follow that link to the molecule's page and the .cdx file will be there, on its own. Don't worry if the molecule is not shown as a picture on the wiki yet - our new data on recent analogs is ahead of the wiki editing.
4. Download the .cdx file for any molecule you need strings for. You do this by clicking on the file, generating a pop up window, then saving the file to your desktop. Don't try to open the file in a browser.
5. Open the .cdx file in a program. Now it looks like you ought to be able to use a free version of Marvin for this (https://www.chemaxon.com/products/marvin/), or maybe an online version, but let us know if this is not possible. We need a way for people without Chemdraw to read those files. This is the step most likely needing a clever workaround, but maybe we'll get lucky.
6. When you've opened the file, select the molecule and do something like "copy as" SMILES. Paste whatever is on the clipboard to some text editor (use whatever) to make sure it looks OK (a bit like the ones already on the wiki) and is free of formatting. Then put the string into the wiki and into the Google Sheet. Do the same for InChI and InChIKey.
7. Repeat for any other molecules lacking strings.
8. Don't worry about potencies for now unless you want to. Some of these are not yet on the wiki.

Have a go at doing two molecules, then ping me here so I can check that things are looking OK. Any issues, write them here. Other onlookers who can offer particular help include @drc007 and @cdsouthan - gents if you can improve this workflow, please say. The aim: to generate the complete set of data for Series 4 compounds in the Google Sheet and, just for completeness, on the wiki.

**Comment by drc007** [comment]
Hi @jlee8294, 

The google docs spreadsheet for editing is here.

https://docs.google.com/spreadsheets/d/1Rvy6OiM291d1GN_cyT6eSw_C3lSuJ1jaR7AJa8hgGsc/edit#gid=510297618

The link @mattodd gave you is to the visualiser.

**Comment by cdsouthan** [comment]
Sheet looks good.  Would be useful  to add the PubChem CIDs and ChEMBL IDs as a columns and complete the InChIKey.  You can use https://pubchem.ncbi.nlm.nih.gov/idexchange/idexchange.cgi for mappings using InChI strings. The stats I got were  (from 120 InChI strings) were chemicalixe.org =  114 (but thats and/or SMILES)   InChI string > PubChem IDs = 100 (as expected, nearly all ChEMBL-only SIDs) InChI string upload conversions = 104 (i.e. 16 failed).  Its somewhat unfortunate there are no synonyms in the CIDs. This means your MMV (exept two from my own CIDs) and OSM numbers are not mapped.  This means you'll have to look after this your end (i.e. in the master sheet) or submit to PubChem yourselves with your own synonym mappings (which has some advantages).

**Comment by mattodd** [comment]
Hi @jlee8294 - any luck?

**Comment by jlee8294** [comment]
The server is down for some odd reason but yes I have (: check on the page?

Sent from my iPhone

On 6 May 2015, at 12:55 pm, Mat Todd <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @jlee8294<https://github.com/jlee8294> - any luck?

-

**Comment by mattodd** [comment]
Oh, good work! So the workflow above is OK - i.e. you're able to download .cdx files and use Marvin to open them and extract the strings, or are you doing something different?

**Comment by jlee8294** [comment]
Actually I've been using chemsketch. Is that alright?

Sent from my iPhone

On 6 May 2015, at 10:26 pm, Mat Todd <notifications@github.com<mailto:notifications@github.com>> wrote:


Oh, good work! So the workflow above is OK - i.e. you're able to download .cdx files and use Marvin to open them and extract the strings, or are you doing something different?

-

**Comment by cdsouthan** [comment]
Why not use Marvin in the first place instead of Chemdraw ?  The free licence is essentialy universal

**Comment by mattodd** [comment]
Well, in this case interoperability is the most important thing - that we
can create and read files in a common format. If chemsketch and Marvin can
read and generate .cdx files, we're safe using Chemdraw (which we use
because we have it). Is there a most widely-employed standard we ought to
be using as a default?
Jessica - whatever works is fine!

On 6 May 2015 at 23:30, cdsouthan <notifications@github.com> wrote:

>  Why not use Marvin in the first place instead of Chemdraw ? The free
> licence is essentialy universal
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-99459555>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by cdsouthan** [comment]
Of course, you team on the gound should use what works and is familliar in the first instance.  I would just say circumspection for the future is necessary as you move from 100s to 1000s of cpds.  The more interconversions and cross-mappings you have to do, the more round-triping noise you bring into the system.

**Comment by drc007** [comment]
This is why I prefer SMILES (or InChi) as the common storage type for molecular structure. If all else fails you can use a plain text editor!

**Comment by jlee8294** [comment]
I suppose it's because I'm more familiar to chemsketch than marvin. But I can switch to using that if that is what you guys want. (:

Sent from my iPhone

On 6 May 2015, at 11:30 pm, cdsouthan <notifications@github.com<mailto:notifications@github.com>> wrote:


Why not use Marvin in the first place instead of Chemdraw ? The free licence is essentialy universal

-

**Comment by mattodd** [comment]
No, don't worry about the means, Jessica. What matters is the end: that you
can generate the strings.

My question is less about the software or string type, but more about the
*file* type for chemical drawings. We will I think always need to visualise
structures offline, e.g. in adding content to the wiki, in writing papers
or in writing blog posts, etc. So when we do that - when we need a
human-readable picture and where it is not possible to generate the picture
automatically - what file type should we share the picture as? I know that
OpenBabel allows conversion, but what should the default be? If chemsketch
and Marvin can read .cdx files, can we use .cdx files, or ought we to use
something else as the default? .mol? .cml? I'm not asking for people's
favourites, just what is seen as being the most widely-used standard.
Looping in @egonw

On 7 May 2015 at 08:00, Jessica Lee <notifications@github.com> wrote:

>  I suppose it's because I'm more familiar to chemsketch than marvin. But I
> can switch to using that if that is what you guys want. (:
>
> Sent from my iPhone
>
> On 6 May 2015, at 11:30 pm, cdsouthan <notifications@github.com<mailto:
> notifications@github.com>> wrote:
>
>
> Why not use Marvin in the first place instead of Chemdraw ? The free
> licence is essentialy universal
>
> -
> Reply to this email directly or view it on GitHub<
> https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-99459555
> >.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-99621732>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by cdsouthan** [comment]
The most comon standard is SDF (or at least PubChem has 68 million). and,  as Chris recomends, SMILES and InChI strings, on a good day, will three-way round-trip with SDF.  As we know there are many choices of redering engines.  For the record Jessica we are not suggesting anything we "want" but merely that OSM cheminformatics becomes (within resource limitations) optimised for your scientific objectives and is consumable by others

**Comment by drc007** [comment]
Matt I suspect you are opening a whole can of worms. As far as I am aware no web browser, or document creation software (other than LaTex) can render chemical structures natively. In the dim and distant past we used plugins and applets to render the structures, but unless the reader had the correct plugin then they still could not render the structure. As a way around this we now insert image files (png, pdf etc.) into documents, web pages and store the chemical information in the meta data associated with the image file. This hopefully means that the user will be able to view the structure but we have very, very little control how the meta data is preserved. As I understand it each software company has their own way of storing the chemical information in meta data and there seems to be no interest in them supporting each others implementation. A long time ago I wrote a tutorial showing a way to add SMILES strings to the meta data associated with images (http://www.macinchem.org/applescript/tutorial9.php) but there has been zero interest in adopting such an approach which might aid interoperability.
My own (perhaps jaundiced) view is that you should regard the rendered image version of structures as simply images, and if you want to preserve chemical information embedded in a document you need to include a text based version e.g. SMILES.

**Comment by mattodd** [comment]
Absolutely - we need the strings, e.g. SMILES. But we're also humans and
need pictures. In the case of complex schemes (e.g.
http://openwetware.org/wiki/Image:Metabolism_Summary.png) we need the page
hosting the image to have the strings, But we *also need the image*. So -
what format should we be sharing, by default, so that others may open and
change the image. I agree there is no effective chemical rendering
mechanism on web pages that works consistently and in a way that is not
annoying. So we need files that people can open and alter. Should the
format be .cdx?

On 7 May 2015 at 16:00, Chris Swain <notifications@github.com> wrote:

>  Matt I suspect you are opening a whole can of worms. As far as I am
> aware no web browser, or document creation software (other than LaTex) can
> render chemical structures natively. In the dim and distant past we used
> plugins and applets to render the structures, but unless the reader had the
> correct plugin then they still could not render the structure. As a way
> around this we now insert image files (png, pdf etc.) into documents, web
> pages and store the chemical information in the meta data associated with
> the image file. This hopefully means that the user will be able to view the
> structure but we have very, very little control how the meta data is
> preserved. As I understand it each software company has their own way of
> storing the chemical information in meta data and there seems to be no
> interest in them supporting each others implementation. A long time ago I
> wrote a tutorial showing a way to add SMILES strings to the meta data
> associated with images (http://www.macinchem.org/applescript/tutorial9.php)
> but there has been zero interest in adopting such an approach which might
> aid interoperability.
> My own (perhaps jaundiced) view is that you should regard the rendered
> image version of structures as simply images, and if you want to preserve
> chemical information embedded in a document you need to include a text
> based version e.g. SMILES.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/290#issuecomment-99732143>
> .
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/chemistry/research/todd.html | W
http://opensourcemalaria.org/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
The short answer is there is no standard. 
Longer answer, whilst chemical structures have a degree of interoperability as soon as you include other layout objects (text, arrows, boxes) then all bets are off.

**Comment by alintheopen** [comment]
I would suggest .cdx as we can open it with more programs...and then the associated strings as a separate entity. I'm with Mat, as chemists we still need pictures as that's how our minds work best!  As an aside, ee should probably also set a standard way of drawing things for the project in different programs. All of this saves time and energy when putting together papers/talks etc. I like the idea of using open software of course but ChemDraw is still our go to in terms of presentation quality. Thanks for your help Jessica!

**Comment by drc007** [comment]
If you are only trying to capture chemical structure information, then sdf, SMILES etc are probably better options. However if you are trying to display pictorial information then .cdx is a reasonable option, just be aware that it may not display correctly in programs other than ChemDraw. Also I suspect that contributors who don't have access to ChemDraw will balk at the cost of an individual license.

**Comment by alintheopen** [comment]
yup, that's why we always need strings as well as a picture, but sometimes pictures speak louder than strings. Let's go with .cdx for now I think.
